### ISSUE TITLE
NPE when doing join on array column

### ISSUE THREAD

**Issue by rmoff** [confluentinc/ksql#4735]
**Describe the bug**

When I run a query it aborts with a NullPointerException

**To Reproduce**
Steps to reproduce the behavior, include:
1. The version of KSQL : 0.7.1
2. Sample source data: n/a
3. Any SQL statements you ran

```
SELECT  WLAN_FC_TYPE_SUBTYPE[1]         AS WLAN_FC_TYPE_SUBTYPE,           
        PT.PACKET_TYPE,                                                    
        WLAN_RADIO_CHANNEL[1]           AS WLAN_RADIO_CHANNEL,             
        WLAN_RADIO_SIGNAL_PERCENTAGE[1] AS WLAN_RADIO_SIGNAL_PERCENTAGE,   
        WLAN_RADIO_SIGNAL_DBM[1]        AS WLAN_RADIO_SIGNAL_DBM,          
        WLAN_RADIO_DURATION[1]          AS WLAN_RADIO_DURATION,            
        WLAN_RA[1]                      AS WLAN_RA,                        
        WLAN_RA_RESOLVED[1]             AS WLAN_RA_RESOLVED,               
        WLAN_DA[1]                      AS WLAN_DA,                        
        WLAN_DA_RESOLVED[1]             AS WLAN_DA_RESOLVED,               
        WLAN_TA[1]                      AS WLAN_TA,                        
        WLAN_TA_RESOLVED[1]             AS WLAN_TA_RESOLVED,               
        WLAN_SA[1]                      AS WLAN_SA,                        
        WLAN_SA_RESOLVED[1]             AS WLAN_SA_RESOLVED,               
        WLAN_STAA[1]                    AS WLAN_STAA,                      
        WLAN_STAA_RESOLVED[1]           AS WLAN_STAA_RESOLVED,             
        WLAN_TAGGED_ALL[1]              AS WLAN_TAGGED_ALL,                
        WLAN_TAG_VENDOR_DATA                                               
        WLAN_TAG_VENDOR_OUI_TYPE                                           
FROM    PCAP P                                                             
          LEFT JOIN                                                        
        WLAN_PACKET_TYPE_REPARTITION PT                                    
          ON P.WLAN_FC_TYPE_SUBTYPE[1]=PT.ROWKEY                           
EMIT CHANGES;
```

**Expected behavior**
The query should work :) 

**Actual behaviour**
A clear and concise description of what actually happens, including:
1. CLI output
2. Error messages

```
+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
|WLAN_FC_|PACKET_T|WLAN_RAD|WLAN_RAD|WLAN_RAD|WLAN_RAD|WLAN_RA |WLAN_RA_|WLAN_DA |WLAN_DA_|WLAN_TA |WLAN_TA_|WLAN_SA |WLAN_SA_|WLAN_STA|WLAN_STA|WLAN_TAG|WLAN_TAG|
|TYPE_SUB|YPE     |IO_CHANN|IO_SIGNA|IO_SIGNA|IO_DURAT|        |RESOLVED|        |RESOLVED|        |RESOLVED|        |RESOLVED|A       |A_RESOLV|GED_ALL |_VENDOR_|
|TYPE    |        |EL      |L_PERCEN|L_DBM   |ION     |        |        |        |        |        |        |        |        |        |ED      |        |OUI_TYPE|
|        |        |        |TAGE    |        |        |        |        |        |        |        |        |        |        |        |        |        |        |
+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
io.confluent.ksql.util.KsqlException
Caused by: java.lang.NullPointerException
Query terminated
ksql>

```
3. KSQL logs

[npe.log](https://github.com/confluentinc/ksql/files/4306007/npe.log)

**Additional context**

```
ksql> DESCRIBE PCAP;

Name                 : PCAP
 Field                        | Type
----------------------------------------------------------
 ROWTIME                      | BIGINT           (system)
 ROWKEY                       | VARCHAR(STRING)  (system)
 TIMESTAMP                    | BIGINT
 WLAN_FC_TYPE_SUBTYPE         | ARRAY<INTEGER>
 WLAN_RADIO_CHANNEL           | ARRAY<INTEGER>
 WLAN_RADIO_SIGNAL_PERCENTAGE | ARRAY<VARCHAR(STRING)>
 WLAN_RADIO_SIGNAL_DBM        | ARRAY<DOUBLE>
 WLAN_RADIO_DURATION          | ARRAY<INTEGER>
 WLAN_RA                      | ARRAY<VARCHAR(STRING)>
 WLAN_RA_RESOLVED             | ARRAY<VARCHAR(STRING)>
 WLAN_DA                      | ARRAY<VARCHAR(STRING)>
 WLAN_DA_RESOLVED             | ARRAY<VARCHAR(STRING)>
 WLAN_TA                      | ARRAY<VARCHAR(STRING)>
 WLAN_TA_RESOLVED             | ARRAY<VARCHAR(STRING)>
 WLAN_SA                      | ARRAY<VARCHAR(STRING)>
 WLAN_SA_RESOLVED             | ARRAY<VARCHAR(STRING)>
 WLAN_STAA                    | ARRAY<VARCHAR(STRING)>
 WLAN_STAA_RESOLVED           | ARRAY<VARCHAR(STRING)>
 WLAN_TAGGED_ALL              | ARRAY<VARCHAR(STRING)>
 WLAN_TAG_VENDOR_DATA         | ARRAY<VARCHAR(STRING)>
 WLAN_TAG_VENDOR_OUI_TYPE     | ARRAY<VARCHAR(STRING)>
----------------------------------------------------------
For runtime statistics and query details run: DESCRIBE EXTENDED <Stream,Table>;
ksql>

ksql> describe WLAN_PACKET_TYPE_REPARTITION;

Name                 : WLAN_PACKET_TYPE_REPARTITION
 Field       | Type
-----------------------------------------
 ROWTIME     | BIGINT           (system)
 ROWKEY      | INTEGER          (system)
 PACKET_TYPE | VARCHAR(STRING)
-----------------------------------------
For runtime statistics and query details run: DESCRIBE EXTENDED <Stream,Table>;
ksql>
```

**Comment by vcrfxia** [comment]
This looks like it could be related to https://github.com/confluentinc/ksql/issues/5481, which has been fixed since 0.9.0.

**Comment by big-andy-coates** [comment]
With out the sample data I can't be certain, but I'd say the most likely cause is that `WLAN_FC_TYPE_SUBTYPE` is null and so the array access is resulting in an NPE when trying to repartition by the `P.WLAN_FC_TYPE_SUBTYPE[1]` expression.

This is already fixed by https://github.com/confluentinc/ksql/pull/4822.
